include/nbl/builtin/hlsl/workgroup/shared_scan.hlsl

```diff
@@ -52,21 +51,20 @@ struct WorkgroupScanHead
         // every group of 64 invocations has been coallesced into 1 result value. This means that the results of 
         // the first SubgroupSz^2 invocations will be processed by the first subgroup and so on.
         // Consequently, those first SubgroupSz^2 invocations will store their results on SubgroupSz scratch slots 
-        // with halfSubgroupSz padding and the next level will follow the same + the previous as an `offset`.
-        const uint offset = (gl_LocalInvocationIndex >> glsl::gl_SubgroupSizeLog2()) & ~subgroupMask; // For subgroupSz = 64, first 4095 invocations get offset 0, 4096-8191 offset 64, then 128 etc.
-        const uint memBegin = offset;
-        uint nextLevelStoreIndex = memBegin + glsl::gl_SubgroupID();
-        scanStoreIndex = gl_LocalInvocationIndex;
+        // and the next level will follow the same + the previous as an `offset`.
+        
+        scanLoadIndex = gl_LocalInvocationIndex;
+        const uint loadStoreIndexDiff = scanLoadIndex - glsl::gl_SubgroupID();
         
         bool participate = gl_LocalInvocationIndex <= lastInvocationInLevel;
         while(lastInvocationInLevel >= glsl::gl_SubgroupSize() * glsl::gl_SubgroupSize())
         {
-            sharedAccessor.main.workgroupExecutionAndMemoryBarrier();
+            // REVIEW-519: Refactor to use ping-ponging if possible in order to minimize barrier usage. Worth it? How common is WGSZ > 4096
```

Comment:
no this is pefect, you already have it down to minimal barriers required, the ping-ponging I meant for a reduction (so you don't progress along geom sequence in offsets - the whole `isScan` thing) but I've given up now